src/VestRewardsDistributionJob.sol

```diff
@@ -0,0 +1,152 @@
+    function unpaid(uint256 _id) external view returns (uint256);
+}
+
+interface VestedRewardsDistributionLike {
+    function distribute() external returns (uint256 amount);
+    function dssVest() external view returns (DssVestWithGemLike);
+    function lastDistributedAt() external view returns (uint256);
+    function vestId() external view returns (uint256);
+}
+
+/// @title Call distribute() when possible
+contract VestRewardsDistributionJob is IJob {
+
+    using EnumerableSet for EnumerableSet.AddressSet;
+
+    // --- Auth ---
+    mapping (address => uint256) public wards;
+
+    function rely(address usr) external auth {
+        wards[usr] = 1;
+
+        emit Rely(usr);
+    }
+    function deny(address usr) external auth {
+        wards[usr] = 0;
+
+        emit Deny(usr);
+    }
+    modifier auth {
+        require(wards[msg.sender] == 1, "VestRewards/not-authorized");
+        _;
+    }
+
+    SequencerLike public immutable sequencer;
+
+
+    EnumerableSet.AddressSet private distributions;
+
+
+    mapping(address => uint256) public distributionIntervals;
+
+    // --- Errors ---
+    error CannotDistributeYet(address rewDist);
+    error NoArgs();
+    error NotMaster(bytes32 network);
+    error RewardDistributionExists(address rewDist);
+    error RewardDistributionDoesNotExist(address rewDist);
+
+    // --- Events ---
+    event Work(bytes32 indexed network, address rewDist, uint distAmounts);
+    event Rely(address indexed usr);
+    event Deny(address indexed usr);
+    event AddRewardDistribution(address indexed rewDist, uint256 interval);
+    event RemoveRewardDistribution(address indexed rewDist);
+    event ModifiedDistributionInterval(address indexed rewDist, uint256 interval);
+
+    constructor(address _sequencer) {
+        sequencer = SequencerLike(_sequencer);
+        wards[msg.sender] = 1;
+        emit Rely(msg.sender);
+    }
+
+    // --- Reward Distribution Admin ---
+    function addRewardDistribution(address rewDist, uint256 interval) external auth {
+        if (!distributions.add(rewDist)) revert RewardDistributionExists(rewDist);
+        distributionIntervals[rewDist] = interval;
+        emit AddRewardDistribution(rewDist, interval);
+    }
+
+    function removeRewardDistribution(address rewDist) external auth {
+        if (!distributions.remove(rewDist)) revert RewardDistributionDoesNotExist(rewDist);
+        delete distributionIntervals[rewDist];
+        emit RemoveRewardDistribution(rewDist);
+    }
+
+    function modifyDistributionInterval(address rewDist, uint256 interval) external auth {
+        if (!distributions.contains(rewDist)) revert RewardDistributionDoesNotExist(rewDist);
+        distributionIntervals[rewDist] = interval;
+        emit ModifiedDistributionInterval(rewDist, interval);
+    }
+
+    function work(bytes32 network, bytes calldata args) public {
+        if (!sequencer.isMaster(network)) revert NotMaster(network);
+        if (args.length == 0) revert NoArgs();
```

Comment:
This check is probably redundant. `abi.decode` below would fail if `args` is empty.

Reply:
it depends, when i was passing an empty address array it returned some weird value. Maybe it just reverts but i don't think we should leave undefined behaviours when we can avoid it.

Reply:
If you pass a valid encoded empty array of addresses, it will return this:

```
➜ address[] memory emptyArr = new address[](0);
➜ abi.encode(emptyArr)
Type: dynamic bytes
├ Hex (Memory):
├─ Length ([0x00:0x20]): 0x0000000000000000000000000000000000000000000000000000000000000040
├─ Contents ([0x20:..]): 0x00000000000000000000000000000000000000000000000000000000000000200000000000000000000000000000000000000000000000000000000000000000
├ Hex (Tuple Encoded):
├─ Pointer ([0x00:0x20]): 0x0000000000000000000000000000000000000000000000000000000000000020
├─ Length ([0x20:0x40]): 0x0000000000000000000000000000000000000000000000000000000000000040
└─ Contents ([0x40:..]): 0x00000000000000000000000000000000000000000000000000000000000000200000000000000000000000000000000000000000000000000000000000000000
```

If you try to decode it as a single address instead of an array of addresses, it will wrongly give you the following:

```
➜ abi.decode(abi.encode(emptyArr), (address))
tuple(address) = (
  ├ Type: address
  └ Data: 0x0000000000000000000000000000000000000020
)
```

However, if you try to decode an empty `bytes` value, you get an error:

```
➜ a = abi.decode("", (address))
Traces:
  [974] 0xBd770416a3345F91E4B34576cb804a576fa48EB1::run()
    └─ ← [Revert] EvmError: Revert

⚒️ Chisel Error: Failed to inspect expression
```

However, being a little bit more explicit is okay in this case. So I'm fine leaving it as it is.